mlrun/runtimes/sparkjob/abstract.py

```diff
@@ -231,18 +230,20 @@ def deploy_default_image(cls, with_gpu=False):
         sj = new_function(kind=cls.kind, name="spark-default-image-deploy-temp")
         sj.spec.build.image = cls._get_default_deployed_mlrun_image_name(with_gpu)
 
-        sj.with_executor_requests(cpu=1, mem="512m", gpus=1 if with_gpu else None)
-        sj.with_driver_requests(cpu=1, mem="512m", gpus=1 if with_gpu else None)
+        sj.with_executor_requests(cpu=1, mem="512m")
+        sj.with_executor_limits(cpu=1, mem="512m", gpus=1 if with_gpu else None)
+        sj.with_driver_requests(cpu=1, mem="512m")
+        sj.with_driver_limits(cpu=1, mem="512m", gpus=1 if with_gpu else None)
```

Comment:
This function is just here to build the image, I'm not sure why it even needs requests or limits set